tests/test_bootloader.py

```diff
@@ -0,0 +1,269 @@
+              as piksi_flash:
+            with Timeout(130) as timeout:
+              piksi_flash.write_ihx(STM_FW)
+
+  def test_flash_nap_firmware(self):
+    """ Test flashing NAP hexfile. """
+    with serial_link.get_driver(use_ftdi=False, port=PORT1) as driver:
+      with Handler(driver.read, driver.write) as link:
+        link.start()
+
+        self.set_btldr_mode(link)
+
+        with Bootloader(link) as piksi_bootloader:
+          with Timeout(10) as timeout:
+            piksi_bootloader.wait_for_handshake()
+          with flash.Flash(link, flash_type='M25', sbp_version=piksi_bootloader.version) \
+              as piksi_flash:
+            with Timeout(250) as timeout:
+              piksi_flash.write_ihx(NAP_FW)
+
+  def test_program_btldr(self):
+    """ Test programming the bootloader once its sector is locked. """
+    with serial_link.get_driver(use_ftdi=False, port=PORT1) as driver:
+      with Handler(driver.read, driver.write) as link:
+        link.start()
+
+        self.set_btldr_mode(link)
+
+        with Bootloader(link) as piksi_bootloader:
+          piksi_bootloader.wait_for_handshake()
+          with flash.Flash(link, flash_type='STM', sbp_version=piksi_bootloader.version) \
+              as piksi_flash:
+            # Make sure the bootloader sector is locked.
+            piksi_flash.lock_sector(0)
+            # Make sure the address to test isn't already programmed.
+            piksi_flash.read(0x08003FFF, 1)
+            waiting_for_read = piksi_flash.get_n_queued_ops() > 0
+            while waiting_for_read:
+              waiting_for_read = piksi_flash.get_n_queued_ops() > 0
+            byte_read = piksi_flash._read_callback_ihx.gets(0x08003FFF, 1)
+            self.assertEqual('\xFF', byte_read,
+                             "Address to program is already programmed")
+            # Attempt to write 0x00 to last address of the sector.
+            piksi_flash.program(0x08003FFF, '\x00')
+            waiting_for_read = piksi_flash.get_n_queued_ops() > 0
+            while waiting_for_read:
+              waiting_for_read = piksi_flash.get_n_queued_ops() > 0
+            byte_read = piksi_flash._read_callback_ihx.gets(0x08003FFF, 1)
+            self.assertEqual('\xFF', byte_read,
+                             "Bootloader sector was programmed")
+
+  def test_erase_btldr(self):
+    """ Test erasing the bootloader once its sector is locked. """
+    pass
+
+  def test_set_btldr_mode_wrong_sender_id(self):
+    """
+    Test setting Piksi into bootloader mode with an incorrect sender ID
+    (should fail).
+    """
+    pass
+
+  """ Test flashing using an incorrect sender ID (should fail). """
+  def test_flashing_wrong_sender_id(self):
+    pass
+
+  """
+  Test if two Piksies can set eachother into bootloader mode (should fail).
+  """
+  def test_two_piksies_btldr_mode(self):
+    if PORT2 is None:
+      return
+
+  """ Test if two Piksies can simultaneously bootload. """
```

Comment:
What's going on with these docstrings up here? Also, if you're planning on filling in these tests later, perhaps add a TODO or `raise NotImplementedError("TODO")`